device/cuda/src/finding/finding_algorithm.cu

```diff
@@ -286,76 +252,111 @@ finding_algorithm<stepper_t, navigator_t>::operator()(
     vecmem::data::vector_buffer<detray::geometry::barcode> barcodes_buffer{
         n_modules, m_mr.main};
 
-    unsigned int nThreads = m_warp_size * 2;
-    unsigned int nBlocks = (barcodes_buffer.size() + nThreads - 1) / nThreads;
+    {
+        const unsigned int nThreads = m_warp_size * 2;
+        const unsigned int nBlocks =
+            (barcodes_buffer.size() + nThreads - 1) / nThreads;
 
-    kernels::make_barcode_sequence<<<nBlocks, nThreads, 0, stream>>>(
-        uniques_buffer, barcodes_buffer);
+        kernels::make_barcode_sequence<<<nBlocks, nThreads, 0, stream>>>(
+            uniques_buffer, barcodes_buffer);
 
-    TRACCC_CUDA_ERROR_CHECK(cudaGetLastError());
+        TRACCC_CUDA_ERROR_CHECK(cudaGetLastError());
+    }
 
-    for (unsigned int step = 0; step < m_cfg.max_track_candidates_per_track;
-         step++) {
+    const unsigned int n_seeds = m_copy.get_size(seeds_buffer);
 
-        // Previous step
-        const unsigned int prev_step = (step == 0 ? 0 : step - 1);
+    // Prepare input parameters with seeds
+    bound_track_parameters_collection_types::buffer in_params_buffer(n_seeds,
+                                                                     m_mr.main);
 
-        // Reset the number of tracks per seed
-        m_copy.memset(n_tracks_per_seed_buffer, 0)->ignore();
+    {
+        bound_track_parameters_collection_types::device in_params(
+            in_params_buffer);
+        bound_track_parameters_collection_types::const_device seeds(
+            seeds_buffer);
+        thrust::copy(thrust::cuda::par.on(stream), seeds.begin(), seeds.end(),
+                     in_params.begin());
+    }
```

Comment:
Why are you using `thrust::copy` here instead of vecmem's copy object? 😕

Reply:
This is how the code already was! I didn't change anything here.

Reply:
We can change it to vecmem::copy for sure

Reply:
Trying to figure out the right syntax for this.